src/wp-includes/global-styles-and-settings.php

```diff
@@ -374,6 +364,45 @@ function ( $item ) {
 	}
 }
 
+/**
+ * Gets the block name from a given theme.json path.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param string $path Path to a property in theme.json.
+ * @return string Identified block name, or empty string if none found.
+ */
+function wp_get_block_name_from_theme_json_path( $path ) {
```

Comment:
Can we keep this function inline in the `wp_add_global_styles_for_blocks`? I'd like to contain the explosion of little utilities like this as public (or even private) functions, if they are only used in one place. Note that this only works for `styles.blocks.blockName`, but we also have `settings.blocks.blockName`, so it's not a general utility.

Reply:
@oandregal I have explicitly marked this function as private, similar to other private functions in core. That alone clarifies that it is not a general utility.

Fair point on the name though, I can make it more verbose.

Having this function standalone facilitates testing, which is crucial to have. The only way to avoid privately marked global functions is to make heavier use of classes, which would require a broader refactoring though, so I think that would go too far in scope for this change. If you prefer that, I'm happy to explore in a separate PR to introduce e.g. some sort of `WP_Block_CSS_Loader` class. I think that would be very helpful to have, to encapsulate logic like this, allow use of actual private methods and also to easily override in Gutenberg, similar to how we do with other `theme.json` related classes. I would have liked to open this PR against Gutenberg, but for this one it wouldn't work. Using a class would have improved the situation.